### PR TITLE
phan fixes 2

### PR DESCRIPTION
**Pull request by peteraba**
Second batch of fixes for [phan](https://github.com/phan/phan) warnings. Note that it's probably better to review this PR commit by commit.

### CODE REVIEW

**Comment by davidbyoung** [comment]
Thanks for the PRs. I don't have anything against Phan, but since it is not currently being used by the framework, I'd prefer not to include any Phan-specific annotations in the code base. Also, it looked like there were some code changes made just to pass analysis (eg testing for array indices that will always be set) - these types of changes feel unnecessary IMO. However, I do like other problems it is picking up on, eg incorrect PHPDoc or legitimate bugs.

**Comment by peteraba** [comment]
I do not disagree, these PRs exist so that you can cherry-pick what you like and what you don't. Obviously I went with my best judgement and tried to be as strict as reasonable. Still, I've disabled one check and excluded a lot of directories from being tested.

That said, the point is that I think [phan](https://github.com/phan/phan) **should be** used by the framework (CI) and already created a PR for it.

File: src/Opulence/Databases/Adapters/Pdo/Connection.php [hunk 1]
```diff
@@ -182,6 +183,9 @@ public function prepare($statement, $driverOptions = [])
     }
 
     /**
+     * @suppress PhanParamSignatureMismatchInternal
```
Comment (davidbyoung) [inline src/Opulence/Databases/Adapters/Pdo/Connection.php]: Lets remove these annotations for now

File: src/Opulence/Databases/ConnectionPools/ConnectionPool.php [hunk 2]
```diff
@@ -201,21 +204,23 @@ protected function getConnection(string $type, Server $server) : IConnection
                     throw new RuntimeException('No master specified');
                 }
 
-                if ($this->servers['master']['connection'] == null) {
-                    $this->servers['master']['connection'] = $this->connectToServer($server);
+                if ($this->servers['master']['connection'] instanceof  IConnection) {
```
Comment (davidbyoung) [inline src/Opulence/Databases/ConnectionPools/ConnectionPool.php]: This feels like an unnecessary change

File: src/Opulence/Framework/Tests/Http/Testing/PhpUnit/Mocks/IntegrationTestCase.php [hunk 3]
```diff
@@ -98,6 +98,7 @@ public function setUp()
     }
 
     /**
+     * @suppress PhanTypeMismatchReturn
```
Comment (davidbyoung) [inline src/Opulence/Framework/Tests/Http/Testing/PhpUnit/Mocks/IntegrationTestCase.php]: Please remove

File: src/Opulence/Framework/Tests/Http/Testing/PhpUnit/Mocks/IntegrationTestCase.php [hunk 4]
```diff
@@ -108,6 +109,7 @@ protected function getExceptionHandler() : IExceptionHandler
...
```
Comment (davidbyoung) [inline src/Opulence/Framework/Tests/Http/Testing/PhpUnit/Mocks/IntegrationTestCase.php]: Please remove

File: src/Opulence/Ioc/Bootstrappers/Caching/FileCache.php [hunk 5]
```diff
@@ -63,11 +63,13 @@ public function get()
         $decodedContents = json_decode($rawContents, true);
         $registry = new BootstrapperRegistry();
 
-        foreach ($decodedContents['eager'] as $eagerBootstrapperClass) {
+        $eager = isset($decodedContents['eager']) ? $decodedContents['eager'] : [];
```
Comment (davidbyoung) [inline src/Opulence/Ioc/Bootstrappers/Caching/FileCache.php]: This also feels like an unnecessary check

File: src/Opulence/Ioc/Bootstrappers/Caching/FileCache.php [hunk 6]
```diff
@@ -63,11 +63,13 @@ public function get()
...
+        foreach ($eager as $eagerBootstrapperClass) {
             $registry->registerEagerBootstrapper($eagerBootstrapperClass);
         }
 
-        foreach ($decodedContents['lazy'] as $boundClass => $bindingData) {
+        $lazy = isset($decodedContents['lazy']) ? $decodedContents['lazy'] : [];
```
Comment (davidbyoung) [inline src/Opulence/Ioc/Bootstrappers/Caching/FileCache.php]: Same thing about feeling unnecessary

File: src/Opulence/Orm/DataMappers/CachedSqlDataMapper.php [hunk 7]
```diff
@@ -263,6 +263,7 @@ protected function scheduleForCacheUpdate($entity)
      * Does the comparison of entities in cache to entities in the SQL database
      * Also performs refresh if the user chooses to do so
      *
+     * @suppress PhanTypeMismatchReturn
```
Comment (davidbyoung) [inline src/Opulence/Orm/DataMappers/CachedSqlDataMapper.php]: Please remove

File: src/Opulence/Orm/Ids/Accessors/IdAccessorRegistry.php [hunk 8]
```diff
@@ -43,6 +43,7 @@ function ($entity, $id) {
     }
 
     /**
+     * @suppress PhanTypeArraySuspicious
```
Comment (davidbyoung) [inline src/Opulence/Orm/Ids/Accessors/IdAccessorRegistry.php]: Please remove

File: src/Opulence/Orm/Ids/Accessors/IdAccessorRegistry.php [hunk 9]
```diff
@@ -102,6 +103,7 @@ public function registerReflectionIdAccessors($classNames, string $idPropertyNam
...
```
Comment (davidbyoung) [inline src/Opulence/Orm/Ids/Accessors/IdAccessorRegistry.php]: Please remove

File: src/Opulence/Orm/Repositories/Repository.php [hunk 10]
```diff
@@ -80,6 +80,7 @@ public function getById($id)
     /**
      * Performs a get query on the data mapper and adds any results as managed entities to the unit of work
      *
+     * @suppress PhanTypeMismatchReturn
```
Comment (davidbyoung) [inline src/Opulence/Orm/Repositories/Repository.php]: Please remove

**Comment by peteraba** [comment]
@davidbyoung tried to fix what you requested. let me know if you'd prefer me to squash the commits into one or if you see any issues remaining.

**Comment by davidbyoung** [comment]
I'll squash and merge them.  Thanks a ton!
